sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Percentile.scala

```diff
@@ -98,7 +92,7 @@ case class Percentile(
   }
 
   override def inputTypes: Seq[AbstractDataType] = percentageExpression.dataType match {
-    case _: ArrayType => Seq(NumericType, ArrayType)
+    case _: ArrayType => Seq(NumericType, ArrayType(DoubleType))
```

Comment:
If the input array's widest type is a `DecimalType`, `CreateArray()` will create a `ArrayType(DecimalType, containsNull = true)`, because the function `Cast. forceNullable(fromType, toType)` always returns true if `toType` is a `DecimalType`.
Then, if we are expecting a inputType as `ArrayType(DoubleType, containsNull = false)`, we fail to cast because the nullability is not compatible.
For example, the `array(0.0, 1.0)` will be casted to `ArrayType(DecimalType(2, 1), containsNull = true)`, and this is a valid input for `Percentile` function.
Thus, I decide to keep the `inputType` to be `ArrayType(DoubleType, containsNull = true)`, to be compatible with the above case.